keps/sig-node/1287-in-place-update-pod-resources/README.md

```diff
@@ -962,20 +1101,18 @@ _This section must be completed when targeting alpha to a release._
 
 * **How can this feature be enabled / disabled in a live cluster?**
   - [x] Feature gate (also fill in values in `kep.yaml`)
-    - Feature gate name: InPlacePodVerticalScaling
+    - Feature gate name: `InPlacePodVerticalScaling`
     - Components depending on the feature gate: kubelet, kube-apiserver, kube-scheduler
-  - [ ] Other
-    - Describe the mechanism:
-    - Will enabling / disabling the feature require downtime of the control
-      plane? No.
-    - Will enabling / disabling the feature require downtime or reprovisioning
-      of a node? No.
 
-* **Does enabling the feature change any default behavior?** No
+* **Does enabling the feature change any default behavior?**
+
+  - Kubelet sets several pod status fields: `AllocatedResources`, `Resources`
 
 * **Can the feature be disabled once it has been enabled (i.e. can we roll back
   the enablement)?** Yes
 
+  - The feature should not be disabled on a running node (create a new node instead).
```

Comment:
Change to "No"?

```suggestion
* **Can the feature be disabled once it has been enabled (i.e. can we roll back
  the enablement)?** No

  - The feature should not be disabled on a running node (create a new node instead).
```

Reply:
BTW, what does "disable on a running node" mean? Can this feature be turned on/off on a single node?

Reply:
TBH, I don't remember why I added this. I don't think it should be a problem to disable it, although there are some potential race conditions around it.

Reply:
This is a PRR ask. Featuregate allows you to enable/disable it at will (but not on per-node granularity). There was one bug around hash calculation that would restart containers even those not using the feature, but that has been fixed. Other races might exist and hopefully get discovered is it bakes alpha/beta pre-prod deployments.

Reply:
OK, I removed the statement that the node needs to be recreated or drained. However, if there are pending or in-progress resizes, it's possible pods could be left in an unknown state, where depending on exactly when the restart happened they could be finished resizing, not resized, or partially resized, and the pod API will not accurately reflect the state.

Reply:
Sounds like there are enough gotchas here that we should retain some of this in the answer.  I asked PRR leads about this situation and one piece of feedback was that we shouldn't assume clusters have disposable nodes, so proscribing that nodes should be drained here would be appropriate, but proscribing that new nodes be created would not be.